src/filesystem.cpp

```diff
@@ -689,44 +718,39 @@ static bfs::path windows_userdata(const std::string& newprefdir)
 		// Custom directory relative to workdir (for portable installs, etc.)
 		dir = get_cwd() + "/" + temp;
 		DBG_FS << "userdata relative to current working directory";
+	} else if(!temp.empty() && temp[0] == '~') {
+		std::optional<std::string> games_path = get_games_path();
+
+		if(games_path) {
+			dir = *games_path;
+		} else {
+			dir = get_cwd();
+		}
+		dir / temp.substr(1);
```

Comment:
Does this somehow assign to `dir`?

Reply:
Fixed in f1b90e91a976ea22990fbd75e85ed52cb8e1857a